File: platforms/ios/devkit/Sources/NativeblocksDevkit/feature/auth/domain/repository/AuthRepository.swift
import Combine

internal protocol AuthRepository {
    func updateTokenAndEndpoint(qrData: String) -> Bool
    func updateTokenAndEndpoint(endpoint: String, token: String, realtimeEndpoint: String) -> Bool
    func authCheck() -> Bool
    func logout()
    func token() -> AnyPublisher<String, Never>
}

File: platforms/android/devkit/devkit/src/main/java/io/nativeblocks/devkit/feature/auth/domain/repository/AuthRepository.kt
package io.nativeblocks.devkit.feature.auth.domain.repository

import kotlinx.coroutines.flow.Flow

internal interface AuthRepository {
    suspend fun updateTokenAndEndpoint(qrData: String): Boolean
    suspend fun updateTokenAndEndpoint(endpoint: String, token: String, realtimeEndpoint: String): Boolean
    suspend fun authCheck(): Boolean
    suspend fun logout()
    fun token(): Flow<String>
}